src/main/java/org/jvnet/hudson/test/QueryUtils.java

```diff
@@ -0,0 +1,96 @@
+package org.jvnet.hudson.test;
+
+import org.htmlunit.html.HtmlElement;
+import org.htmlunit.html.HtmlPage;
+
+import java.util.concurrent.TimeUnit;
+
+public final class QueryUtils {
+
+    private QueryUtils() {}
+
+    /**
+     * Waits until the given string is visible on the page, otherwise throws an exception
+     * @param page the page
+     * @param value the value to find
+     * @throws RuntimeException if string is not present after three seconds
+     */
+    public static void waitUntilStringIsPresent(HtmlPage page, String value) {
+        long maxWaitTime = TimeUnit.SECONDS.toMillis(3);
+        long startTime = System.currentTimeMillis();
+
+        while (System.currentTimeMillis() - startTime < maxWaitTime) {
+            if (page.querySelector("*").getTextContent().contains(value)) {
```

Comment:
can probably be simplified with `asText()`? https://stackoverflow.com/a/6538127/4951015

Reply:
Couldn't find the `asText` method, `getVisibleText` exists however so I've updated it with that